### DISCUSSION TITLE
What's the equivalent of StartCoroutine?

### DISCUSSION THREAD

**Discussion by TMPxyz** [Cysharp/UniTask#505] [Q&A]
Hi, 

Is it possible to start an async unitask from a sync method like using StartCoroutine(method)?

It seems the UniTask loop resides in the unity events sequence somewhere, is it possible to push an async method into that?

**Comment by timcassell** [comment]
Unlike Coroutines, async methods start immediately when you call them.

**Reply by TMPxyz** [reply]
Yep, we could just push a async method to Unitask by using `xxx().Forget()`;

`async UniTaskVoid xxx(){ ... }`

btw, coroutine does start immediately until the first yield statement, no?

**Reply by timcassell** [reply]
> coroutine does start immediately until the first yield statement, no?

No, it starts when you pass it to `StartCoroutine`. It pauses at `yield return`, similar to how async methods pause at `await`.
